iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/CertificateBasedAuthenticationCheck.java

```diff
@@ -90,8 +91,24 @@ private static void checkLinkedServices(ContextualResource resource) {
     }
   }
 
-  private static boolean isBooleanFalse(Tree tree) {
-    return ((ArmTree) tree).is(ArmTree.Kind.BOOLEAN_LITERAL) && !((BooleanLiteral) tree).value();
+  private static void checkWebSites(ContextualResource resource) {
+    resource.property("clientCertEnabled")
+      .reportIf(isBooleanFalse(), DISABLED_CERTIFICATE_MESSAGE)
+      .reportIfAbsent(MISSING_CERTIFICATE_MESSAGE);
+    resource.property("clientCertMode")
+      .reportIf(isValue(str -> !"Required".equals(str)), ALLOWING_NO_CERTIFICATE_MESSAGE)
+      .reportIfAbsent(MISSING_CERTIFICATE_MESSAGE);
+  }
+
+  private static void checkWebSitesSlots(ContextualResource resource) {
+    resource.property("clientCertEnabled")
+      .reportIf(isBooleanFalse(), DISABLED_CERTIFICATE_MESSAGE);
+    resource.property("clientCertMode")
+      .reportIf(isValue(str -> !"Required".equals(str)), ALLOWING_NO_CERTIFICATE_MESSAGE);
+  }
+
+  private static Predicate<Expression> isBooleanFalse() {
```

Comment:
1. Maybe `isFalse()` a better name? (I'm not sure, to be honest).
2. Maybe this should be moved somewhere to be more accessible to other checks? I can imagine that it can be useful in other checks.

Reply:
👍 for the `isFalse()`
I'm tempted to keep it like this for now and when needed later somewhere else we create a class to hold such methods.